publishers/community/generic.py

```diff
@@ -65,11 +65,11 @@ def _delete_dictionary_fields(publication, regexp):
         next_item = fringe.popleft()
 
         if isinstance(next_item, dict):
-            for key in next_item.keys():
+            for key in list(next_item.keys()):
```

Comment:
did 2to3 do this? casting to a list creates a copy, which I'm unsure why that would be necessary here

Reply:
oh, I'm thinking it's because the dictionary is mutated during iteration, in which case that makes sense and you can ignore me :)